src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1380,51 +1399,79 @@ LedgerMaster::findNewLedgersToPublish(
     auto valLedger = mValidLedger.get();
     std::uint32_t valSeq = valLedger->info().seq;
 
+    // Create a range of ledgers we need to publish
+    RangeSet<std::uint32_t> toPublish;
+    toPublish.insert(range(pubSeq, valSeq));
+
+    // IMPORTANT: This RangeSet subtraction is critical for standalone mode when
+    // loading large catalogue files (millions of ledgers). Standalone mode is
+    // NOT just for tests - it's used for loading catalogue packs without
+    // consensus stealing resources. Without this optimization, tryAdvance would
+    // attempt to publish ALL pinned ledgers, keeping them and their SHAMap
```

Comment:
TODO: standalone mode would need to be configured to use actual database files for standalone mode to actually work